api/src/main/java/ai/djl/modality/audio/AudioFactory.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package ai.djl.modality.audio;
+
+import ai.djl.ndarray.NDArray;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * {@code AudioFactory} contains audio creation mechanism on top of different platforms like PC and
+ * Android. System will choose appropriate Factory based on the supported audio type.
+ */
+public abstract class AudioFactory {
+
+    private static final Logger logger = LoggerFactory.getLogger(AudioFactory.class);
+
+    @SuppressWarnings({"PMD.DoubleBraceInitialization", "PMD.UseConcurrentHashMap"})
+    private static final Map<String, Boolean> FACTORIES =
+            new LinkedHashMap<String, Boolean>() {
+                {
+                    put("ai.djl.audio.FFmpegAudioFactory", true);
+                    put("ai.djl.modality.audio.SampledAudioFactory", false);
+                }
+            };
+
+    protected Configuration conf;
+
+    protected AudioFactory(Configuration conf) {
+        this.conf = conf;
+    }
+
+    /**
+     * Create new instance of audio factory from the provided factory implementation.
+     *
+     * @param conf The configuration for {@link AudioFactory} to use.
+     * @return {@link AudioFactory}
+     */
+    public static AudioFactory newInstance(Configuration conf) {
+        boolean android = "http://www.android.com/".equals(System.getProperty("java.vendor.url"));
```

Comment:
I think we should just a simple try and fail approach for both Android and PC

Reply:
OK